src/components/ArticleViewer.jsx

```diff
@@ -1,26 +1,51 @@
 import { useParams } from "react-router-dom"
-import { getArticle } from '../api'
+import { getArticle, patchVotes } from '../api'
 import { useEffect, useState } from "react";
 import Expandable from "./Expandable";
 import CommentsList from "./CommentsList";
 import formatDate from "../utilities/formatDate";
 const ArticleViewer = () => {
-    const [currArticle, setCurrArticle] = useState({})
-    const [isLoading, setIsLoading] = useState(true)
+    const [currArticle, setCurrArticle] = useState({});
+    const [isLoading, setIsLoading] = useState(true);
+    const [voteCount, setVoteCount] = useState(0);
+    const [error, setError] = useState(null);
     const { article_id } = useParams();
     useEffect(() => {
         getArticle(article_id)
             .then((article) => {
                 setCurrArticle(article)
+                setVoteCount(article.votes)
             })
             .catch((err) => {
                 //err handling
             })
             .finally(() => {
                 setIsLoading(false)
             })
-    }, [currArticle]);
-    const date = new Date(currArticle.created_at);
+    }, []);
+
+    const handleVote = (event) => {
+        const errorMsg = "Your vote was unsuccessful, please try again."
+        if (event.target.value === "up") {
+            setVoteCount((currVoteCount) => currVoteCount + 1);
+            setError(null)
+            patchVotes(article_id, { inc_votes: 1 })
+                .catch((err) => {
+                    setVoteCount((currVoteCount) => currVoteCount - 1);
+                    setError(errorMsg)
+                })
+        } else if (event.target.value === "down"){
+
+            setVoteCount((currVoteCount) => currVoteCount - 1);
+        setError(null)
+        patchVotes(article_id, { inc_votes: -1 })
+            .catch((err) => {
+                setVoteCount((currVoteCount) => currVoteCount + 1);
+                setError(errorMsg)
+            })
```

Comment:
Instead of checking that the value is up or down, if you passed it 1 or -1 then you could just pass that to a single version of patchVotes that accepts whatever that value is! Then you wouldn't have two blocks of code that are very similar 

Good job catering for the error in both cases here though! If you refactor again you could use that single param that is passed to handleVote :)